app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -59,13 +59,48 @@ def self.storage_inv_to_hashes(inv)
         :master              => storage_inv[:master]
       }
 
+      new_result[:disks] = storage_inv_to_disk_hashes(storage_inv, new_result)
+
       result << new_result
       result_uids[mor] = new_result
       result_uids[:storage_id][storage_inv[:id]] = new_result
     end
     return result, result_uids
   end
 
+  def self.storage_inv_to_disk_hashes(inv, storage)
```

Comment:
Again I believe breaking this into 2 methods will be clearer, for me its seems hard to understand what inventory exactly is being processed here, it would be nice to have a method that has a name that explains what are this inventory item.
And then another method that builds the result.